dvc/remote/s3.py

```diff
@@ -273,4 +273,11 @@ def _generate_download_url(self, path_info, expires=3600):
 
     def walk_files(self, path_info, max_items=None):
         for fname in self._list_paths(path_info, max_items):
+            if self._is_empty_dir(fname):
+                continue
+
             yield path_info / posixpath.relpath(fname, path_info.path)
+
+    @staticmethod
+    def _is_empty_dir(fname):
+        return fname.endswith("/")
```

Comment:
I would inline it, will be easier to read. And the name of the method is confusing anyway.

Reply:
@Suor , what do you mean by inline it :thinking: ?

something like this?

```python
    def walk_files(self, path_info, max_items=None):
        def is_empty_dir(fname): return fname.endswith("/")

        for fname in self._list_paths(path_info, max_items):
            if is_empty_dir(fname):
                continue

            yield path_info / posixpath.relpath(fname, path_info.path)
```

What method name would you put it? I named it like that because make sense when reading the whole loop: `for fname in list_paths, if fname is an empty directory then skip it`

Reply:
@mroutis He means using `fname.endswith("/")` instead of `_is_empty_dir(fname)`

Reply:
ok, thanks, @efiop